tox.ini

```diff
@@ -1,28 +1,56 @@
 [tox]
-envlist = py35, py36, docs
+envlist = py35, py36, docs, lint, pylint
 skip_missing_interpreters = True
 
-[tox:travis]
-3.5 = py35
-3.6 = py36, docs
-
 [testenv]
 deps =
   -r{toxinidir}/requirements_test.txt
-commands = pytest --timeout=2 --cov=pyhap --cov-report= {posargs:pyhap tests}
+commands =
+  pytest --timeout=2 --cov=pyhap --cov-report= {posargs:pyhap tests}
 
 [testenv:temperature]
 basepython = python3.6
 deps =
   -r{toxinidir}/requirements_all.txt
-commands = python main.py
+commands =
+  python main.py
 
 [testenv:docs]
 changedir = docs
-deps = -r{toxinidir}/requirements_docs.txt
+deps =
+  -r{toxinidir}/requirements_docs.txt
 commands =
   make clean
   sphinx-build -W -b html source {envtmpdir}/html
 whitelist_externals=
   /usr/bin/make
   make
+
+
+[testenv:lint]
+basepython = {env:PYTHON3_PATH:python3}
+deps =
+  -r{toxinidir}/requirements_test.txt
+commands =
+  flake8 pyhap tests --ignore=D10,D205,D4,E501 --exclude=pyhap/accessories/*
```

Comment:
The ignores should be removed once the docs pass.
Also the `--exclude` flag can be removed with #115.